bigquery_etl/docs/derived_datasets/generate_derived_dataset_docs.py

```diff
@@ -0,0 +1,72 @@
+"""Generate documentation for derived datasets."""
+
+import os
+import yaml
+from pathlib import Path
+from jinja2 import Environment, FileSystemLoader
+
+VIEW_FILE = "view.sql"
+METADATA_FILE = "metadata.yaml"
+NON_USER_FACING_DATASET_SUFFIXES = (
+    "_derived",
+    "_external",
+    "_bi",
+    "_restricted",
+)
+SOURCE_URL = "https://github.com/mozilla/bigquery-etl/blob/master"
+
+
+def generate_derived_dataset_docs(out_dir, project_dir):
+    """Generate documentation for derived datasets."""
+    project_doc_dir = Path(out_dir) / "mozdata"
+
+    # get a list of all user-facing datasets
+    data_sets = [
+        item
+        for item in os.listdir(project_dir)
+        if os.path.isdir(os.path.join(project_dir, item))
+        and all(name not in item for name in NON_USER_FACING_DATASET_SUFFIXES)
+    ]
+
+    for table in data_sets:
+        output = []
+        source_urls = {}
+        with open(project_doc_dir / f"{table}.md", "w") as dataset_doc:
+            # Manually set title to prevent Mkdocs from removing
+            # underscores and capitalizing file names
+            # https://github.com/mkdocs/mkdocs/issues/1915#issuecomment-561311801
+            dataset_doc.write(f"---\ntitle: {table}\n---\n\n")
+
+            for root, dirs, files in os.walk(Path(project_dir) / table):
+                # show views in an alphabetical order
+                dirs.sort()
```

Comment:
Does this actually do anything? The logic seems like it just skips over the directories if they exist.

Reply:
You mean `dirs.sort()`? It sorts the directories that we're looping so when we extract info from root and files things will be parsed to the template in an alphabetical order

Reply:
Ah I see, it's mutating shared state. It wasn't obvious to me at first glance.